crates/snapbox/src/path.rs

```diff
@@ -181,8 +183,11 @@ impl PathDiff {
                     let expected = crate::Data::read_from(&expected_path, None)
                         .map(|d| d.map_text(crate::utils::normalize_lines))
                         .map_err(Self::Failure)?;
-                    if expected.as_str().is_some() {
-                        actual = actual.try_text().map_text(crate::utils::normalize_lines);
+
+                    if expected.format() != DataFormat::Binary {
```

Comment:
Consitional shouldn't be needed here, its all handled with `try_coerce(expected.format())`